### ISSUE TITLE
Autofix for prefer-template + no-useless-concat gives weird output

### ISSUE THREAD

**Issue by remcohaszing** [eslint/eslint#8417]
**Tell us about your environment**

* **ESLint Version:** v3.19.0
* **Node Version:** v7.8.0
* **npm Version:** 4.4.4

**What parser (default, Babel-ESLint, etc.) are you using?** default

**Please show your full configuration:**

<details>
<summary>.eslintrc.yaml</summary>

```yaml
root: true

extends: airbnb

env:
  browser: true
  commonjs: true

settings:
  react:
    version: '15.3'
```
</details>

**What did you do? Please include the actual source code causing the issue.**

1. I wrote the following line:

   ```js
   `foo` + '/' + bar;
   ```

2. I then saved the file, making the Atom ESLint plugin autofix my code.

**What did you expect to happen?**

I expected the line to become like this:

```js
`${foo}/${bar}`;
```

**What actually happened? Please include the actual, raw output from ESLint.**

When ESLint autofixed this, it became the following:

```js
`${'foo' + '/'}${bar}`;
```

This still triggers no-useless-concat and automatic fixes won't fix this line when it's run again.

**Comment by not-an-aardvark** [comment]
I think this is working as intended because `no-useless-concat` doesn't currently support autofixing. So the only thing that gets fixed is your concatenation of `'/'` and `bar`, while the other concatenation remains unfixed.

**Comment by not-an-aardvark** [comment]
Closing because this is working as intended.
